### PR TITLE
Serverless housekeeping: Local testing guide, small docs fixes, and some handler tests for errors

### PR DESCRIPTION
**Pull request by gregfurman**
Some small fixes/improvements:
- Added some serverless handler tests
- Included short guide on local testing AWS `bento-lamba` using LocalStack 
- Added note on batches failing when exceptions are not caught (from https://github.com/warpstreamlabs/bento/issues/59#issuecomment-2195654914)

### CODE REVIEW

File: website/docs/guides/serverless/lambda.md [hunk 1]
```diff
@@ -109,6 +107,9 @@ output:
           reject: "processing failed due to: ${! error() }"
 ```
 
+**Note: if you are using [partial batch responses](https://docs.aws.amazon.com/lambda/latest/dg/services-sqs-errorhandling.html#services-sqs-batchfailurereporting) then 
+throwing an exception will cause the entire batch to be considered a failure**.
+
```
Comment (jem-davies) [inline website/docs/guides/serverless/lambda.md:112]:
```suggestion
:::caution
Note: if you are using [partial batch responses](https://docs.aws.amazon.com/lambda/latest/dg/services-sqs-errorhandling.html#services-sqs-batchfailurereporting) then 
throwing an exception will cause the entire batch to be considered a failure.
:::
```

Can do this for increased visibility.

File: website/docs/guides/serverless/lambda.md [hunk 2]
```diff
@@ -194,6 +195,67 @@ go build github.com/warpstreamlabs/bento/cmd/serverless/bento-lambda
...
+### Using the AWS CLI v2
+
+First, configure some stubbed environment variables:
+```sh
+export AWS_ACCESS_KEY_ID="test"
+export AWS_SECRET_ACCESS_KEY="test"
+export AWS_DEFAULT_REGION="us-east-1"
+```
+
+To run commands against LocalStack, just [configure the endpoint URL](https://docs.localstack.cloud/user-guide/integrations/aws-cli/#configuring-an-endpoint-url) `--endpoint-url=<localstack-url>` flag in CLI calls to point to LocalStack.
```
Comment (jem-davies) [inline website/docs/guides/serverless/lambda.md:221]:
```suggestion
First, set your AWS CLI to connect to LocalStack:
```sh
export AWS_ACCESS_KEY_ID="test"
export AWS_SECRET_ACCESS_KEY="test"
export AWS_DEFAULT_REGION="us-east-1"
export AWS_ENDPOINT_URL=http://localhost:4566
```
```

File: website/docs/guides/serverless/lambda.md [hunk 3]
```diff
@@ -194,6 +195,67 @@ go build github.com/warpstreamlabs/bento/cmd/serverless/bento-lambda
...
+```sh
+LAMBDA_ENV=`cat yourconfig.yaml | jq -csR {Variables:{BENTO_CONFIG:.}}`
+aws lambda create-function \
+  --runtime go1.x \
+  --handler bento-lambda \
+  --role 'arn:aws:iam::000000000000:role/bento-example-role' \
+  --zip-file fileb://bento-lambda.zip \
+  --environment "$LAMBDA_ENV" \
+  --function-name bento-example \
+  --endpoint-url http://localhost:4566
```
Comment (jem-davies) [inline website/docs/guides/serverless/lambda.md:239]:
```suggestion
```

File: website/docs/guides/serverless/lambda.md [hunk 4]
```diff
@@ -194,6 +195,67 @@ go build github.com/warpstreamlabs/bento/cmd/serverless/bento-lambda
...
+```
+
+Invocation can be done the same way:
+```sh
+aws lambda invoke \                                                       
+  --function-name bento-example \
+  --cli-binary-format raw-in-base64-out \
+  --payload '{"your":"document"}' \
+  --endpoint-url http://localhost:4566 \
```
Comment (jem-davies) [inline website/docs/guides/serverless/lambda.md:248]:
```suggestion
```

Comment (jem-davies) [reply website/docs/guides/serverless/lambda.md:248]: you can do it this way - not a big deal but I think it just cuts down slightly on amount of text.

File: website/docs/guides/serverless/lambda.md
Comment (jem-davies) [inline website/docs/guides/serverless/lambda.md:1]: don't think I can add a comment to a line that hasn't been altered, 

One line 158 there is a mistake `amd64` -> `arm64`?

File: internal/serverless/handler_test.go [hunk 5]
```diff
@@ -202,3 +202,103 @@ output:
 		t.Error(err)
 	}
 }
+
+func TestHandlerError(t *testing.T) {
+	// TODO: Should this not be failing since an exception is thrown?
```
Comment (jem-davies) [inline internal/serverless/handler_test.go:207]: Looks right to me - is testing the behaviour [here](http://localhost:3000/bento/docs/guides/serverless/lambda#processing-errors)

It is throwing an error in the pipeline.processor part of the config so what we should see is: 

 - not see an error 
 - get the final state of the message

```suggestion
```

**Review by jem-davies — CHANGES_REQUESTED** [review]
looks good to me - just a few nitpicky things - feel free to ignore them if you think it better how it is 👍

File: website/docs/guides/serverless/lambda.md (hunk 2 above)
Comment (gregfurman) [reply website/docs/guides/serverless/lambda.md:221]: I like using the environment variable instead. I'll make a comment that it should be unset though before wanting to actually create a resource in AWS.

**Review by jem-davies — APPROVED** [review]

**Pull request merged**